src/benchmarks/micro/libraries/System.Text.RegularExpressions/Perf.Regex.Industry.cs

```diff
@@ -0,0 +1,300 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Extensions;
+using MicroBenchmarks;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    internal static class Perf_Regex_Industry
+    {
+        public static string ReadInputFile(string name)
+        {
+            string path = Path.Combine(Path.GetDirectoryName(typeof(Perf_Regex_Industry).Assembly.Location), "libraries", "System.Text.RegularExpressions", "TestData", name);
```

Comment:
I looked around for a proposal for static methods to decompress from a file to a Span<byte> or something, but only found https://github.com/dotnet/runtime/issues/44793 which still requires you to read the file. Maybe it's not a common thing.

Reply:
Even if we added one, we would be unlikely to use it here, given that an explicit goal of this repo is to be able to run the tests against older versions.

Reply:
BTW, I tried recompressing with maximum compression gzip and it only saves about 10% fwiw...